ui/utils.js

```diff
@@ -232,6 +232,22 @@ function suggest_element_until_click(selector, state_to_set) {
     });
 }
 
+function format_asset_title_for_ui(asset) {
+    let symbol, str;
+    if(asset == 'IOTA') {
+        symbol = 'MIOTA';
+    } else {
+        symbol = asset;
+    }
+    let path = settings.ICON_MAP_LIST.find(x => x.symbol === symbol.toLowerCase()).path;
```

Comment:
as I said above, this is now a list and you are searching it for each icon, which is really slow. This should be a map.